c7n/schema.py

```diff
@@ -69,11 +72,34 @@ def validate(data, schema=None):
         logging.exception(
             "specific_error failed, traceback, followed by fallback")
 
+    extra_errs = validate_parse(data)
+    if extra_errs:
+        return extra_errs
+
     return list(filter(None, [
         errors[0],
         best_match(validator.iter_errors(data)),
     ]))
 
+def validate_parse(data):
```

Comment:
let's be more explicit on the name, and the validation being performed, validate_parse is a bit generic, ie. validate_policy_dates/validate_start_stop, also ideally this would be part of policy.validate() for proper encapsulation.